resources/js/Pages/Collections/Components/CollectionVoting/NominationDialog.tsx

```diff
@@ -49,17 +51,47 @@ const NominationDialogFooter = ({
 export const NominationDialog = ({
     isOpen,
     setIsOpen,
-    collections,
+    initialCollections,
     user,
 }: {
     isOpen: boolean;
     setIsOpen: (isOpen: boolean) => void;
-    collections: VoteCollectionProperties[];
+    initialCollections: VoteCollectionProperties[];
     user: App.Data.UserData | null;
 }): JSX.Element => {
     const { t } = useTranslation();
     const [query, setQuery] = useState<string>("");
     const [selectedCollection, setSelectedCollection] = useState<number>(0);
+    const [collections, setCollections] = useState(initialCollections);
+
+    const [debouncedQuery] = useDebounce(query, 500);
+
+    const searchCollections = async (): Promise<void> => {
+        if (debouncedQuery.trim().length === 0) {
+            setCollections(initialCollections);
+
+            return;
+        }
+
+        const { data } = await axios.get<{
+            collections: VoteCollectionProperties[];
+        }>(route("nominatable-collections", { query: debouncedQuery }));
```

Comment:
Hey @crnkovic, I know it is in draft yet, I'd use `use-query` instead of axios, that would make it much easier.